src/framework-ondata.js

```diff
@@ -1,18 +1,298 @@
+
+             // Don't have a wrapper fRunUpdater 
+             throw Error( 'No wrapper fRunUpdater function found!  Cannot run updater.' );
+
+         } else { 
+             // TODO: revisit this logic later, per rdf-pipeline/noflo-rdf-pipeline#35
+
+             // save data used to see if we have a different state after running updater
+             var lastErrorState = _.clone( vni.errorState() );  // Shallow clone 
+             var lastOutputLm = vni.outputState().lm;
+ 
+             clearStateData( vni.errorState ); // clear last error state
+
+             // Save wrapper to make it accessible from the Promise
+             var wrapper = this.nodeInstance.wrapper;
+
+             new Promise(function( resolve ) { 
+ 
+                 // Execute fRunUpdater which will also execute the updater
+                 resolve( wrapper.fRunUpdater.call( vni ) );
+
+             }).then( function() { 
+                 // fRunUpdater/Updater success path
+             
+                 // Returned OK, but updater could have set an error - check for that
+                 if ( stateChange( vni.errorState, lastErrorState ) ) {
+                    setOutputErrorFlag( vni );   
+                    lastOutputLm = undefined;  // got a new error so we will force sending output
+                 }
+
+                 // If not new output or error state data, send it downstream or log it to the console
+                 handleOutput( outputPorts.output, lastOutputLm, vni.outputState() );
+                 handleError( outputPorts.error,  lastErrorState.lm, vni.errorState() );
+
+             }, function( rejected ) { 
+                 // fRunUpdater/updater failed 
+                 if ( isInitState( vni.errorState ) ) { 
+                     // fRunUpdater/updater has not already set an error state - use the rejected info
+                     changeStateData( vni.errorState, rejected );
+                 }
+
+                 setOutputErrorFlag( vni, true ); 
+
+                 stateChange( vni.outputState, lastOutputLm );
+                 stateChange( vni.errorState, lastErrorState );
+
+                 handleOutput( outputPorts.output, lastOutputLm, vni.outputState() );
+                 handleError( outputPorts.error, lastErrorState.lm, vni.errorState() );
+
+                 // If we haven't already processed the rejected error, do it now
+                 if ( rejected !== vni.errorState().data ) { 
+                     lastErrorState = _.clone( vni.errorState() );   
+                     changeStateData( vni.errorState, rejected );
+                     if ( changeStateData( vni.errorState, lastErrorState ) ) { 
+                         lastErrorState.lm = undefined; 
+                     } 
+                     handleError( outputPorts.error, lastErrorState.lm, vni.errorState() );
+                 }
+
+             }).catch( function( e ) { 
+                 // fRunUpdater or Updater threw an error
```

Comment:
This would not catch an error from fRunUpdate or Updater (directly) only errors from the function starting on line 75. Change this comment to something like // Could not handle output or error result